hcl2template/formatter.go

```diff
@@ -135,7 +136,8 @@ func (f *HCL2Formatter) processFile(filename string) ([]byte, error) {
 
 	_, diags := f.parser.ParseHCL(inSrc, filename)
 	if diags.HasErrors() {
-		return nil, fmt.Errorf("failed to parse HCL %s", filename)
+		errs := multierror.Append(fmt.Errorf("failed to parse HCL %s", filename), diags.Errs()...)
+		return nil, errs
```

Comment:
Since we already display  `"encountered an error while formatting %s"` what if we drop the "failed to parse HCL %s" error message as well. 

```suggestion
		return nil, errors.New(diags.Error())
```

So instead of 
```shell
~>  packer fmt command/test-fixtures/fmt_errs/malformed.pkr.hcl
Error: encountered an error while formatting command/test-fixtures/fmt_errs/malformed.pkr.hcl

2 errors occurred:
        * failed to parse HCL command/test-fixtures/fmt_errs/malformed.pkr.hcl
        * command/test-fixtures/fmt_errs/malformed.pkr.hcl:5,1-8: Argument or block
definition required; An argument or block definition is required here. To set an
argument, use the equals sign "=" to introduce the argument value.
```

We get 
```shell
~>  packer fmt command/test-fixtures/fmt_errs/malformed.pkr.hcl

Error: encountered an error while formatting command/test-fixtures/fmt_errs/malformed.pkr.hcl

command/test-fixtures/fmt_errs/malformed.pkr.hcl:5,1-8: Argument or block
definition required; An argument or block definition is required here. To set an
argument, use the equals sign "=" to introduce the argument value.
```

Reply:
Done, just to be clear, `diags.Error()` returns a list of errors, so either we use multierror to encompass it as one error, or we have to essentially build an error message manually for this.
I think this is a valid use case for using `multierror`, so that's what I went with here, without the extra "failed to parse HCL" which doesn't really help understanding/fixing the error.